apps/src/p5lab/spritelab/blocks.js

```diff
@@ -199,7 +199,18 @@ const customInputTypes = {
         .appendTitle(new Blockly.FieldImage('', 32, 32), inputConfig.name);
     },
     generateCode(block, arg) {
-      return `'${block.getTitleValue(arg.name)}'`;
+      switch (block.type) {
+        case 'gamelab_clickedSpritePointer':
+          return '{id: extraArgs.sprite}';
+        case 'gamelab_subjectSpritePointer':
+          return '{id: extraArgs.sprite}';
```

Comment:
For PR documentation: When I first read this, I was expecting to see three different IDs (i.e. `extraArgs.sprite` is here twice). Why is the `.sprite` ID reused?

Reply:
When we detect an event, we call back into the interpreted code, passing in any relevant info in the `extraArgs` parameter. For click events, there's just one sprite, so the shape of that object is `{sprite: <some id>}`. For touch events, there are two sprites, so we have `{sprite: <some id>, target: <some id>}`
The generated code for the clicked/subject/object pointer blocks just need to evaluate to the appropriate sprite, referenced by its id, so the generated code should be `{id: <some id>}`. In the case of click events, the relevant id is passed in as `extraArgs.sprite` and in the case of touch events, the ids are `extraArgs.sprite` and `extraArgs.target`